src/components/DataTable/index.tsx

```diff
@@ -27,6 +28,7 @@ type DataTableProps<Type> = {
   canExpand?: boolean,
   accordionRenderer?: (c: Type) => ReactNode,
   showTopBarOnEmpty?: boolean
+  variant: 'default' | 'grid' | 'striped-column' | 'striped-row',
```

Comment:
here also. Also make the variants a type of itself and use it everywhere which requires it